File: projector/typescript/src/opts.ts
import cli from "command-line-args";

export type ProjectorOptions = {
    arguments?: string[];
    pwd?: string;
    config?: string;
};

export default function getOptions(): ProjectorOptions {
    return cli([
        {
            name: "arguments",
            type: String,
            defaultOption: true,
            multiple: true,
        },
        {
            name: "pwd",
            type: String,
            alias: "p",
        },
        {
            name: "config",
            type: String,
            alias: "c",
        },
    ]) as ProjectorOptions;
}

File: projector/rust/src/opts.rs
use clap::Parser;
use std::path::PathBuf;

#[derive(Parser, Debug)]
#[clap()]
pub struct ProjectorOptions {
    pub arguments: Vec<String>,

    #[clap(short = 'p', long = "pwd")]
    pub pwd: Option<PathBuf>,

    #[clap(short = 'c', long = "config")]
    pub config: Option<PathBuf>,
}
